sdks/python/apache_beam/metrics/cells.py

```diff
@@ -238,6 +277,74 @@ def mean(self):
     return float(self.data.sum)/self.data.count
 
 
+class GaugeResult(object):
+  def __init__(self, data):
+    self.data = data
+
+  def __eq__(self, other):
+    return self.data == other.data
+
+  def __ne__(self, other):
+    return not self.__eq__(other)
+
+  def __repr__(self):
+    return '<GaugeResult(value={}, timestamp={})>'.format(
+        self.value,
+        self.timestamp)
+
+  @property
+  def value(self):
+    return self.data.value
+
+  @property
+  def timestamp(self):
+    return self.data.timestamp
+
+
+class GaugeData(object):
+  """For internal use only; no backwards-compatibility guarantees.
+
+  The data structure that holds data about a gauge metric.
+
+  Gauge metrics are restricted to integers only.
+
+  This object is not thread safe, so it's not supposed to be modified
+  by other than the GaugeCell that contains it.
+  """
+  def __init__(self, value, timestamp=None):
+    self.value = value
+    self.timestamp = timestamp if timestamp is not None else time.time()
```

Comment:
<!--new_thread; commit:3b2d3da5e97094e4521c80e07118d80343634698; resolved:0-->
Can we do the `time.time()` initialization elsewhere?  It would be preferable for data classes to not depend on anything external, if possible.